code/modules/clothing/spacesuits/rig/modules/ninja.dm

```diff
@@ -205,3 +208,108 @@
 		holder.wearer.drop_from_inventory(src)
 		qdel(holder)
 	qdel(src)
+
+/obj/item/rig_module/emp_shielding
+	name = "EMP dissipation module"
+	desc = "A bewilderingly complex bundle of fiber optics and chips. Seems like it uses a good deal of power."
+	active_power_cost = 10
+	toggleable = 1
+	usable = 0
+
+	activate_string = "Enable Active EMP Shielding"
+	deactivate_string = "Disable Active EMP Shielding"
+
+	interface_name = "active EMP shielding system"
+	interface_desc = "A highly experimental system that augments the hardsuit's existing EM shielding."
+	var/protection_amount = 30
+
+/obj/item/rig_module/emp_shielding/activate()
+	if(!..())
+		return
+
+	holder.emp_protection += protection_amount
+
+/obj/item/rig_module/emp_shielding/deactivate()
+	if(!..())
+		return
+
+	holder.emp_protection = max(0,(holder.emp_protection - protection_amount))
+
+/obj/item/rig_module/emergency_powergenerator
+	name = "emergency power generator"
+	desc = "A high yield power generating device that takes a long time to recharge."
+	active_power_cost = 0
+	toggleable = 0
+	usable = 1
+	confined_use = 1
+	var/cooldown = 0
+
+	engage_string = "Use Emergency Power"
+
+	interface_name = "emergency power generator"
+	interface_desc = "A high yield power generating device that takes a long time to recharge."
+	var/generation_ammount = 1500
+
+/obj/item/rig_module/emergency_powergenerator/engage()
+	if(!..())
+		return
+	var/mob/living/carbon/human/H = holder.wearer
+	if(cooldown)
+		H << "<span class='danger'>There isn't enough power stored up yet!</span>"
+		return 0
+	else
+		H << "<span class='danger'>Your suit emits a loud sound as power is rapidly injected into your suits battery!</span>"
+		playsound(H.loc, 'sound/effects/sparks2.ogg', 50, 1)
+		holder.cell.give(generation_ammount)
+		cooldown = 1
+		addtimer(CALLBACK(src, /obj/item/rig_module/emergency_powergenerator/proc/reset_cooldown), 240)
+
+/obj/item/rig_module/emergency_powergenerator/proc/reset_cooldown()
+	cooldown = 0
+
+/obj/item/rig_module/emag_hand
```

Comment:
I am a bit concerned about this, since this would give the ninja a emag with unlimited usage.
With the traditional emag, you have to think about when you use it, as you can only use it a few times.

I think a better implementation would be to limit the times it can be used initially (i.e. 6 times)
And have it slowly recharge in the background (i.e. 1 usage every 20 minutes)
That would give them a total of 12 emag usages over the course of two hours.

Reply:
I disagree. The emag for traitor's is a tactical tool because traitor's cannot teleport around like the ninja does. While a hefty recharge time is perhaps apt, 12 usages over two hours (which is an inaccurate way to portray it, because there'd possibly 20 minute stretches with 0 usages), seems a little excessive.